samza-core/src/main/java/org/apache/samza/zk/ZkBarrierForVersionUpgrade.java

```diff
@@ -61,9 +61,19 @@
   private final Optional<ZkBarrierListener> barrierListenerOptional;
 
   public enum State {
```

Comment:
Document all valid state-transitions allowed in the state-machine and the events that trigger them. 
Call out that currently, the only allowed transitions are from NEW->DONE and NEW->TIMED_OUT.